File: test-case/common/intbreak/Intbreak.cs
using Soda.Unittest;

public class Solution {
    private int[] memo;

    public int IntegerBreak(int n) {
        memo = new int[59];
        return solve(n);
    }

    private int solve(int n) {
        if (n == 1) {
            return 1;
        }
        if (memo[n] > 0) {
            return memo[n];
        }
        int res = 0;
        for (int i = 1; i < n; ++i) {
            res = Math.Max(i * (n-i), res);
            res = Math.Max(i * solve(n-i), res);
        }
        memo[n] = res;
        return res;
    }
}

public class Intbreak
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().IntegerBreak));
        // work.SetValidator((e, r) => ...);
        work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.readStdin()));
    }
}

File: test-case/common/intbreak/intbreak.ts
import {
    AvlTree,
    BinarySearchTree
} from '@datastructures-js/binary-search-tree';
import { TestWork, Utils } from 'soda/unittest';

// step [1]: implement solution function
let memo: number[] = [];
/**
 * @param {number} n
 * @return {number}
 */
function integerBreak(n: number): number {
    memo = new Array<number>(59).fill(0);
    return solve(n);
}

function solve(n: number): number {
    if (n == 1) {
        return 1;
    }
    if (memo[n] > 0) {
        return memo[n];
    }
    let res = 0;
    for (let i = 1; i < n; ++i) {
        res = Math.max(i * (n-i), i * solve(n-i), res);
    }
    memo[n] = res;
    return res;
}

// step [2]: setup function/return/arguments
let taskFunc = integerBreak;
const work = TestWork.create<ReturnType<typeof taskFunc>>(taskFunc);
// work = TestWork.forStruct(STRUCT);
// work.validator = (x, y) => { ... };
work.compareSerial = true;
console.log(work.run(Utils.fromStdin()));
